File: app/src/main/java/com/katkov/lolachievements/di/provider/LoggingInterceptorProvider.java
package com.katkov.lolachievements.di.provider;

import javax.inject.Inject;
import javax.inject.Provider;

import okhttp3.Interceptor;
import okhttp3.logging.HttpLoggingInterceptor;

public final class LoggingInterceptorProvider implements Provider<Interceptor> {

    @Inject
    LoggingInterceptorProvider() {
    }

    @Override
    public Interceptor get() {
        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        return loggingInterceptor;
    }
}

File: app/src/main/java/com/katkov/lolachievements/di/provider/LoggingInterceptorProvider.kt
package com.katkov.lolachievements.di.provider

import okhttp3.Interceptor
import okhttp3.logging.HttpLoggingInterceptor
import javax.inject.Inject
import javax.inject.Provider

class LoggingInterceptorProvider
@Inject
internal constructor() : Provider<Interceptor> {

    override fun get(): Interceptor {
        val loggingInterceptor = HttpLoggingInterceptor()
        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY
        return loggingInterceptor
    }
}
